### PR TITLE
fix getDockerHostIpAddress within docker container

### PR DESCRIPTION
**Pull request by qoomon**
only determine default gateway within docker container if docker host differ from default, otherwise it was set manually and it should be used as it was set

### CODE REVIEW

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 1]
```diff
@@ -17,42 +17,43 @@
     // See https://github.com/docker/docker/blob/a9fa38b1edf30b23cae3eade0be48b3d4b1de14b/daemon/initlayer/setup_unix.go#L25
     public static final boolean IN_A_CONTAINER = new File("/.dockerenv").exists();
 
-    @Getter(lazy = true)
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: Any reason why you dropped lazily cached result? This change will make it run every time somebody queries the host ip, and I don't see how does it fix it.
Also, was it broken?

Comment (qoomon) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: my bad you are right i will refactor it to lazy cache again.
and yes it is broken if we run it in GitLab CI directly in the build container with the DinD service it resolves to the default gateway, however it should resolve to tcp://docker:2375 or just docker

Comment (bsideup) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: oh wow, nice finding!

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 2]
```diff
@@ -18,41 +18,42 @@
...
-                            return loggingCallback.toString();
-                        } catch (Exception e) {
-                            log.warn("Can't parse the default gateway IP", e);
-                            return null;
-                        }
-                    }
-            ))
-            .map(StringUtils::trimToEmpty)
-            .filter(StringUtils::isNotBlank);
+    private static final Optional<String> defaultGateway = Optional
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: I know that the indentation was incorrect, but could you please revert the change to the actual change? We will fix the indentation afterwards, but it will help to understand what was modified :)

Comment (bsideup) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: I think the whole change is as simple as `getDefaultGateway().orElse("localhost")` in `getDockerHostIpAddress`

Comment (qoomon) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: okay will fixed

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 3]
```diff
@@ -18,10 +18,8 @@
     public static final boolean IN_A_CONTAINER = new File("/.dockerenv").exists();
 
     @Getter(lazy = true)
-    private static final Optional<String> detectedDockerHostIp = Optional
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: Since this getter is public, I don't think we can just change the behaviour of it :( Could you please make this change backward compatible?

Comment (qoomon) [reply core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: ah I see

**Comment by qoomon** [comment]
so finally it is backward compatible so `getDetectedDockerHostIp()` acts as before 
and I marked `getDetectedDockerHostIp()` with `@deprecated`

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 4]
```diff
@@ -31,28 +29,31 @@
...
                         }
                     }
-            ))
-            .map(StringUtils::trimToEmpty)
-            .filter(StringUtils::isNotBlank);
+            ));
+
+    @Deprecated
+    @Getter(lazy = true)
+    private static final Optional<String> detectedDockerHostIp = IN_A_CONTAINER ? getDefaultGateway() : Optional.empty();
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: could you please put this method upper where it was, so that the diff will be smaller?

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 5]
```diff
@@ -31,28 +29,31 @@
...
                             loggingCallback.awaitCompletion(3, SECONDS);
-                            return loggingCallback.toString();
+                            return StringUtils.trimToNull(StringUtils.trim(loggingCallback.toString()));
                         } catch (Exception e) {
                             log.warn("Can't parse the default gateway IP", e);
                             return null;
                         }
                     }
-            ))
-            .map(StringUtils::trimToEmpty)
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: TBH I would prefer to keep this chain

**Comment by qoomon** [comment]
no the diff look quite neat

**Comment by bsideup** [comment]
@qoomon merged, thanks!

**Comment by bsideup** [comment]
@qoomon oh snap, I forgot the changelog :D Will add it myself, thanks for your contribution 👍
